sshconnect.c

```diff
@@ -392,6 +392,9 @@ ssh_create_socket(struct addrinfo *ai)
 		}
 		memcpy(&bindaddr, res->ai_addr, res->ai_addrlen);
 		bindaddrlen = res->ai_addrlen;
+#ifdef IP_BIND_ADDRESS_NO_PORT
+		setsockopt(sock, SOL_IP, IP_BIND_ADDRESS_NO_PORT, &(int) {1}, sizeof(int));
+#endif
 	} else if (options.bind_interface != NULL) {
```

Comment:
The same option is useful if binding to an interface as well, could you rewrite so that both cases are handled?

Reply:
let me check that later.